.github/actions/java-test/action.yaml

```diff
@@ -54,8 +54,13 @@ runs:
     - name: Run tests
       shell: bash
       run: |
-        SPARK_HOME=`pwd` ./mvnw -B clean install ${{ inputs.maven_opts }}
-
+        MAVEN_OPTS="-XX:+UnlockDiagnosticVMOptions -XX:+ShowMessageBoxOnError -XX:+HeapDumpOnOutOfMemoryError -XX:ErrorFile=./hs_err_pid%p.log" SPARK_HOME=`pwd` ./mvnw -B clean install ${{ inputs.maven_opts }}
+    - name: Upload crash logs
+      if: failure()
+      uses: actions/upload-artifact@v4
+      with:
+        name: crash-logs
+        path: hs_err_pid*.log
```

Comment:
Do you think we could also upload `unit-tests.log`?

Reply:
That's a good idea